howso/hierarchy.amlg

```diff
@@ -811,36 +811,101 @@
 		(call !Return (assoc payload (assoc "count" (size cases_to_move)) ))
 	)
 
-	;method to update the references for contained trainees, should only be used for debugging purposes.
-	;{idempotent (true)}
-	#set_contained_trainee_maps
+	;method to update the references for contained trainees
+	;{idempotent (true) protected (true)}
+	#set_hierarchy_relationships
 	(declare
 		(assoc
+			;{type ["string" "null"]}
```

Comment:
This can just be "string" I believe

Reply:
Its possible that we may allow moving a child trainee to be a new root, in which case we would want to set it to null